### PR TITLE
LOG-3122: In Vector, add tracing of the negotiated cipher suite for TLS connections

### PR DESCRIPTION
**Pull request by syedriko**
JIRA:
- https://issues.redhat.com/browse/LOG-3122

### CODE REVIEW

**Comment by syedriko** [comment]
@vimalk78 Check this out, see if it's doing what you need.

File: patch/hyper-openssl/src/lib.rs [hunk 1]
```diff
@@ -233,7 +234,10 @@ where
             let mut stream = SslStream::new(ssl, conn)?;
 
             match Pin::new(&mut stream).connect().await {
-                Ok(()) => Ok(MaybeHttpsStream::Https(stream)),
+                Ok(()) => {
+                    trace!("TLS: current cipher: {}", stream.ssl().current_cipher().unwrap().name());
```
[the remark this answers is not in the document]
Comment (vimalk78) [reply patch/hyper-openssl/src/lib.rs:238]: is this tracing always enabled? shouldn't this tracing be controlled by some flag? will this info be logged in vector logs?

[the remark this answers is not in the document]
Comment (syedriko) [reply patch/hyper-openssl/src/lib.rs:238]: > shouldn't we trace it before we get connection result? line 235?

This is the actual ciphersuite negotiated as part of the handshake, we don't know it before we're done handshaking.

[the remark this answers is not in the document]
Comment (syedriko) [reply patch/hyper-openssl/src/lib.rs:238]: > is this tracing always enabled? shouldn't this tracing be controlled by some flag? will this info be logged in vector logs?

Nope, tracing is not always on. I wrote this up in the JIRA, but it won't hurt to dup it here as well:


In order to enable cipher suite trace, Vector tracing at the "trace" level needs to be enabled for the hyper-openssl crate, with the VECTOR_LOG environment variable. Either set it to "trace" for all components:

`VECTOR_LOG=trace`

or select individual components, but make sure hyper-openssl=trace is included:

`VECTOR_LOG=hyper-openssl=trace`

Vector logs the tracing settings in effect on start up, in a line that looks like so (for VECTOR_LOG=trace):

`2022-09-28T21:13:05.522658Z  INFO vector::app: Log level is enabled. level="trace"`

In the trace output, look for a line like this:

`2022-09-28T20:39:09.735382Z TRACE hyper_openssl: TLS: current cipher: ECDHE-RSA-CHACHA20-POLY1305`

[the remark this answers is not in the document]
Comment (syedriko) [reply patch/hyper-openssl/src/lib.rs:238]: >will this info be logged in vector logs?
Nope, we don't have trace level logs enabled in the vector image.

**Comment by vimalk78** [comment]
@syedriko we need a cherry-pick of this commit on v0.23-rh also

**Pull request merged**